atlp-rwanda/e-comm-team-techsmith-fn
# #185218938 A buyer should be able to checkout

## What does this PR do?

This PR enables a buyer to validate their order and perform checkout to proceed to payment.

## Description of Task to be completed?

+ Given a buyer is logged into their account. They should be able to checkout a product.
+ Given a buyer checks out a product, the order is validated and updates the inventory
+ Given a buy checks out successfully, they are redirected to the payment portal to finalize purchase

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout ft-buyer-checkout-185218938`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

- Check out the components documentation: `npm run storybook`

## Any background context you want to provide?

The checkout button will call the `GET Single Order: '/single/:orderId'` to obtain the order details, and it is the payment button that will call the `POST Checkout Order: '/:id/checkout'`. This is because the `checkout` endpoint automatically performs payment (this is how we built it on the backend).

## What are the relevant pivotal tracker/Trello stories?

+ [#185218938](https://www.pivotaltracker.com/story/show/185218938)

## Screenshots (if applicable)
<img width="1440" alt="Screenshot 2023-06-22 at 03 45 52" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/b8225017-12df-496f-9eee-3f23f826c637">
<img width="1440" alt="Screenshot 2023-06-22 at 03 49 09" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/3b3c95a9-95fb-4ecd-b6da-8dabc9643742">
<img width="1440" alt="Screenshot 2023-06-22 at 04 06 51" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/c595a744-5c46-415c-9589-4cae7009f840">
<img width="1440" alt="Screenshot 2023-06-22 at 04 18 43" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/5fbfad60-43a9-4c7d-8540-d04525475d89">
<img width="1440" alt="Screenshot 2023-06-22 at 04 19 36" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/c441de4e-361a-486b-8234-931d59b94b3f">
